npm/webpack-dev-server/test/e2e.spec.ts

```diff
@@ -87,48 +85,38 @@ describe('#startDevServer', () => {
       },
     })
 
-    return new Promise((res) => {
-      devServerEvents.on('dev-server:compile:success', () => {
-        close(() => res())
-      })
-    })
+    await once(devServerEvents, 'dev-server:compile:success')
+    await close()
   })
 
   it('emits dev-server:compile:error event on error compilation', async () => {
     const devServerEvents = new EventEmitter()
-
     const exitSpy = sinon.stub()
 
+    const badSpec = `${root}/test/fixtures/compilation-fails.spec.js`
     const { close } = await startDevServer({
       webpackConfig,
       options: {
         config,
         specs: [
           {
-            name: `${root}/test/fixtures/compilation-fails.spec.js`,
-            relative: `${root}/test/fixtures/compilation-fails.spec.js`,
-            absolute: `${root}/test/fixtures/compilation-fails.spec.js`,
+            name: badSpec,
+            relative: badSpec,
+            absolute: badSpec,
           },
         ],
         devServerEvents,
       },
     }, exitSpy as any)
 
-    exitSpy()
+    const [err] = await once(devServerEvents, 'dev-server:compile:error')
 
-    return new Promise((res) => {
-      devServerEvents.on('dev-server:compile:error', (err: string) => {
-        if (webpackDevServerFacts.isV3()) {
```

Comment:
This assertion was failing for me under webpack-dev-server v3 - it seemed to work exactly like v4 in my testing, and the CI tests seem to agree. :woman_shrugging:

Reply:
If we're having some per-version stability questions, we need to have system tests that scaffold up a project of that type.

Reply:
Yeah this test-deps.js sucks a lot, this is one of the first things I did at Cypress and didn't know about the system tests. Now I know.

We should not refactor it here though, we have an entire ticket dedicated to it here: https://cypress-io.atlassian.net/browse/UNIFY-1264. I can assign this to @BlueWinds when we work on this project if you'd like to do it.

Reply:
I've no particular attachment to it, but I'm as good a person as any to take that up.